### PR TITLE
drivers/sdcard_spi

### PR DESCRIPTION
**Pull request by kYc0o**
While compiling tests/driver_sdcard_spi with clang in OS X it discovers several minor issues.

This PR fixes them.

### CODE REVIEW

File: drivers/sdcard_spi/sdcard_spi.c [hunk 1]
```diff
@@ -373,7 +373,7 @@ static inline bool _wait_for_not_busy(sdcard_spi_t *card, int32_t max_retries)
 
     do {
         if (_dyn_spi_rxtx_byte(card, SD_CARD_DUMMY_BYTE, &read_byte) == 1) {
-            if (read_byte == 0xFF) {
+            if (read_byte == (char)0xFF) {
```
[the remark this answers is not in the document]
Comment (kYc0o) [reply drivers/sdcard_spi/sdcard_spi.c]: Yes I prefer that too. Will change.

**Comment by kYc0o** [comment]
While fixing @gebart comment I found an extensive use of `char*` in the code to specify buffers. AFAIR @miri64 has already pointed out this to get a consensus on which is better to use, either `char*` or `uint8_t*`, and the consensus was to use `uint8_t*` if I'm not wrong. Therefore, for this file I guess I should modify all the usages of `char*` to `uint8_t*`. However, I don't know if that change is in the scope of this PR.

I don't mind to do it here, just want opinions.

**Comment by MichelRottleuthner** [comment]
I'd vote for simply adding the `char` to `uint8_t` fixes to this PR as a lot of the changes you introduced here are for fixing datatypes anyway.

**Comment by kYc0o** [comment]
So @gebart what's your opinion on this?

**Comment by gebart** [comment]
@kYc0o I'm OK if you want to change the buffer pointer types from char to uint8_t (for this driver and test only) in this PR.
You need to set a better title for the PR.

**Comment by PeterKietzmann** [comment]
What is this PR waiting for?

**Comment by kYc0o** [comment]
My actions, I have other things in the queue but I'll work on this ASAP.

**Comment by kYc0o** [comment]
Well apparently this was fixed by #7919 but following another approach. Should we close this and leave it as is or a more clean solutions should be pursued here?

**Comment by PeterKietzmann** [comment]
@kYc0o it seems there are no strong opinions to have this solution in. I'll close this one now in favor of #7919. New PRs can always be opened, as usual...

**Pull request closed without merging**